jenkins/checkout.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import subprocess
+
+
+def Checkout(git, org, repo, branch, pull):
+    if bool(branch) ^ bool(pull) == False:
```

Comment:
`if bool(branch) == bool(pull):` seems nicer to me, and I would update the error to say "Must specify exactly one of ...".